### ISSUE TITLE
Harmony Exception during init for DSP 0.7.18.6931

### ISSUE THREAD

**Issue by CaffeinatedCM** [hubastard/nebula#308]
Error during patching

DSP Version: ` 0.7.18.6931`

```
[Info   :   BepInEx] Loading [Nebula - Multiplayer Mod 0.0.2.0]
[Info   :Nebula - Multiplayer Mod] Patching Dyson Sphere Program...
[Info   :Nebula - Multiplayer Mod] Applying patches from assembly: NebulaPatcher, Version=0.1.0.0, Culture=neutral, PublicKeyToken=null
[Error  :Nebula - Multiplayer Mod] Unhandled exception occurred while patching the game:
[Error  :Nebula - Multiplayer Mod] HarmonyLib.HarmonyException: IL Compile Error (unknown location) ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
  at System.ThrowHelper.ThrowArgumentOutOfRangeException (System.ExceptionArgument argument, System.ExceptionResource resource) [0x00029] in <2fa7a6a452ca43df998f07fd1486c0df>:0 
  at System.ThrowHelper.ThrowArgumentOutOfRangeException () [0x00000] in <2fa7a6a452ca43df998f07fd1486c0df>:0 
  at HarmonyLib.CodeMatcher.AddLabelsAt (System.Int32 position, System.Collections.Generic.IEnumerable`1[T] labels) [0x00018] in <116dd996dda44480a48a0fc9ee2308ed>:0 
  at HarmonyLib.CodeMatcher.CreateLabelAt (System.Int32 position, System.Reflection.Emit.Label& label) [0x00011] in <116dd996dda44480a48a0fc9ee2308ed>:0 
  at NebulaPatcher.Patches.Transpilers.StationComponent_Transpiler.<ILSUpdateShipPos>g__Transpiler|12_0 (System.Collections.Generic.IEnumerable`1[T] instructions, System.Reflection.Emit.ILGenerator il) [0x007d6] in <0b72fcb96d764b61b72ceacde5965f47>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke(System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00032] in <2fa7a6a452ca43df998f07fd1486c0df>:0 
   --- End of inner exception stack trace ---
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0004b] in <2fa7a6a452ca43df998f07fd1486c0df>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in <2fa7a6a452ca43df998f07fd1486c0df>:0 
  at HarmonyLib.Internal.Patching.ILManipulator.ApplyTranspilers (System.Reflection.Emit.ILGenerator il, System.Reflection.MethodBase original, System.Func`2[T,TResult] getLocal, System.Func`1[TResult] defineLabel) [0x00093] in <116dd996dda44480a48a0fc9ee2308ed>:0 
  at HarmonyLib.Internal.Patching.ILManipulator.WriteTo (Mono.Cecil.Cil.MethodBody body, System.Reflection.MethodBase original) [0x00066] in <116dd996dda44480a48a0fc9ee2308ed>:0 
  at HarmonyLib.PatchFunctions+<>c__DisplayClass3_0.<ReversePatch>b__1 (MonoMod.Cil.ILContext ctx) [0x0015b] in <116dd996dda44480a48a0fc9ee2308ed>:0 
  at MonoMod.Cil.ILContext.Invoke (MonoMod.Cil.ILContext+Manipulator manip) [0x00087] in <43bfc4c6532a47f6b28212e11323aa80>:0 
  at MonoMod.RuntimeDetour.ILHook+Context.InvokeManipulator (Mono.Cecil.MethodDefinition def, MonoMod.Cil.ILContext+Manipulator cb) [0x00012] in <33dff776d87748ecb372a90c5ddfc734>:0 
  at DMD<Refresh>?347076352._MonoMod_RuntimeDetour_ILHook+Context::Refresh (MonoMod.RuntimeDetour.ILHook+Context this) [0x000ea] in <69a076743ce94d599f45965f4283c90b>:0 
  at DMD<>?347076352.Trampoline<MonoMod.RuntimeDetour.ILHook+Context::Refresh>?1662817280 (System.Object ) [0x00020] in <f602021dddd14e09828644411249c138>:0 
  at HarmonyLib.Internal.RuntimeFixes.StackTraceFixes.OnILChainRefresh (System.Object self) [0x00000] in <116dd996dda44480a48a0fc9ee2308ed>:0 
  at MonoMod.RuntimeDetour.ILHook.Apply () [0x00059] in <33dff776d87748ecb372a90c5ddfc734>:0 
  at HarmonyLib.PatchFunctions.ReversePatch (HarmonyLib.HarmonyMethod standin, System.Reflection.MethodBase original, System.Reflection.MethodInfo postTranspiler, System.Reflection.MethodInfo postManipulator) [0x00158] in <116dd996dda44480a48a0fc9ee2308ed>:0 
   --- End of inner exception stack trace ---
  at HarmonyLib.PatchClassProcessor.ReportException (System.Exception exception, System.Reflection.MethodBase original) [0x00045] in <116dd996dda44480a48a0fc9ee2308ed>:0 
  at HarmonyLib.PatchClassProcessor.Patch () [0x00084] in <116dd996dda44480a48a0fc9ee2308ed>:0 
  at HarmonyLib.Harmony.<PatchAll>b__11_0 (System.Type type) [0x00007] in <116dd996dda44480a48a0fc9ee2308ed>:0 
  at HarmonyLib.CollectionExtensions.Do[T] (System.Collections.Generic.IEnumerable`1[T] sequence, System.Action`1[T] action) [0x00014] in <116dd996dda44480a48a0fc9ee2308ed>:0 
  at HarmonyLib.Harmony.PatchAll (System.Reflection.Assembly assembly) [0x00006] in <116dd996dda44480a48a0fc9ee2308ed>:0 
  at NebulaPatcher.NebulaPlugin.InitPatches () [0x00044] in <0b72fcb96d764b61b72ceacde5965f47>:0 
[Info   :Nebula - Multiplayer Mod] Applying Nebula behaviours..
[Info   :Nebula - Multiplayer Mod] Behaviours applied.
```

I was able to get it to work by reverting to the 0.6.x branch of DSP, so I'm guessing they changed something in the recent patch

**Comment by Mushroom** [comment]
This is being addressed in #307,  there is a lot of work to get it functional again

**Comment by sp00ktober** [comment]
This one can be closed, it is possible now to get host and client into the same game on current 0.7.x version.
There are a lot of other things we need to fix tho to have it working like it does for 0.6.x
